inc/alert.class.php

```diff
@@ -240,86 +240,137 @@ public static function findAllToNotify($params = [])
         }
-
-        $query = "SELECT DISTINCT `$table`.`id`, `$table`.*
-                  FROM `$table`
-                  LEFT JOIN `$utable`
-                     ON `$utable`.`plugin_news_alerts_id` = `$table`.`id`
-                     AND `$utable`.`users_id` = $users_id
-                     AND `$utable`.`state` = $hidstate
-                  INNER JOIN `$ttable`
-                     ON `$ttable`.`plugin_news_alerts_id` = `$table`.`id`
-                  $targets_sql
-                  WHERE ($login_show_hidden_sql $login_sql $show_central_sql $show_helpdesk_sql)
-                     AND (`$table`.`date_start` < '$today'
-                           OR `$table`.`date_start` = '$today'
-                           OR `$table`.`date_start` IS NULL
-                     )
-                     AND (`$table`.`date_end` IS NULL
-                           OR `$table`.`date_end` > '$today'
-                           OR `$table`.`date_end` = '$today'
-                     )
-                  AND `is_deleted`='0' AND `is_active`='1'
-                  $entity_sql";
-        $iterator = $DB->request($query);
-        if ($iterator->numrows() < 1) {
+        $criteria = [
+            'SELECT' => ["$table.*"],
+            'DISTINCT' => true,
+            'FROM'   => $table,
+            'LEFT JOIN' => [
+                $utable => [
+                    'ON' => [
+                        $utable => 'plugin_news_alerts_id',
+                        $table  => 'id',
+                        [
+                            'AND' => [
+                                "$utable.users_id" => $users_id,
+                                "$utable.state"    => $hidstate,
+                            ]
+                        ]
+                    ]
+                ]
+            ],
+            'INNER JOIN' => [
+                $ttable => [
+                    'ON' => [
+                        $ttable => 'plugin_news_alerts_id',
+                        $table  => 'id'
+                    ]
+                ]
+            ],
+            'WHERE' => [
+                [$login_show_hidden_sql],
+                [
+                    'OR' => [
+                        ["$table.date_start" => ['<', $today]],
+                        ["$table.date_start" => $today],
+                        ["$table.date_start" => null],
+                    ]
+                ],
+                [
+                    'OR' => [
+                        ["$table.date_end" => ['>', $today]],
+                        ["$table.date_end" => $today],
+                        ["$table.date_end" => null],
+                    ]
+                ],
+                'is_deleted' => 0,
+                'is_active'  => 1,
+            ]
+        ];
+        if (!empty($targets_sql)) {
+            $criteria['INNER JOIN'][$ttable]['ON'][] = $targets_sql;
+        }
+        if (!empty($entity_sql)) {
+            $criteria['WHERE'][] = [$entity_sql];
```

Comment:
No need to encapsulate it, it is already an array.
```suggestion
            $criteria['WHERE'][] = $entity_sql;
```